---
title: 'Effects of inequality on genetics: evidence from UK Biobank'
author: David Hugh-Jones & Abdel Abdellaoui
date: "`r Sys.Date()`"
output: 
  bookdown::pdf_document2:
    toc: false
    latex_engine: xelatex
editor_options: 
  chunk_output_type: console
header-includes:
  - \usepackage{subfig}
  - \usepackage{amsmath}
  - \usepackage{amsthm}
  - \usepackage{setspace}
  - \usepackage{placeins}
  - \newtheorem{prop}{\protect\propositionname}
  - \providecommand{\propositionname}{Proposition}
---

```{r setup, include = FALSE}
library(Formula)
library(car)
library(drake)
library(dplyr)
library(magrittr)
library(AER)
library(ggplot2)
library(Hmisc)
library(huxtable)
library(broom)
library(fixest)
library(santoku)
library(purrr)

knitr::opts_chunk$set(echo = FALSE)

drake::loadd(mf_pairs)
drake::loadd(mf_pairs_twice)
drake::loadd(famhist)
drake::loadd(resid_scores)

famhist %<>% left_join(resid_scores, by = "f.eid")
rm(resid_scores)

famhist$EA3 <- famhist$EA3_excl_23andMe_UK_resid


theme_set(theme_minimal())
```

# Introduction

Charles Murray (1995) warned of "a merging of the cognitive elite with the
affluent". On the opposite side of the political spectrum, Karl Marx (1844)
wrote "I am ugly, but I can buy the most beautiful woman.... the effect of
ugliness, its repelling power, is destroyed by money." These quotations suggest
that social advantages, such as wealth, caste or status, may be transformed into
biological advantages in the next generation, via assortative mating between
socially and genetically advantaged people. We call this process 
*genetic lock-in*.

Figures \@ref(fig:pic-basic-corr-uni) and \@ref(fig:pic-basic-corr-income) 
illustrate the idea using data for spouse
pairs from UK Biobank. Figures \@ref(fig:pic-basic-corr-uni) plots one partner's 
mean polygenic score for educational attainment (PSEA) against a measure of the 
other partner's actual educational
attainment: possession of a university degree. University graduates had spouses
with higher PSEA.^[To minimize concerns about genetic stratification, i.e.
correlations between genetics and non-genetic forms of inherited advantage, PSEA
is residualized by the first 100 principal components of UK Biobank array data.]
Figure \@ref(fig:pic-basic-corr-income) plots one partner's PSEA against another
measure of social status: income. 

These figures do not prove that genetic lock-in is taking place: since an
individual's own PSEA correlates with both their educational attainment, and
their income, both figures could be a result of partner selection on a purely
genetic basis. In this paper, we test the theory more rigorously, using
environmental shocks to social status that are unlikely to be correlated with
own genetics. First, we develop a simple theory of genetic lock-in, to
illustrate how its effects vary with social structure.


```{r pic-basic-corr-uni, fig.subcap = c("Female PSEA by male educational attainment", "Male PSEA by female educational attainment"), fig.cap = "Social and genetic advantage among spouse pairs in UK Biobank", fig.width = 3, fig.height = 3}

pic_ss <- stat_summary(fun.data = mean_cl_normal, na.rm = TRUE)
pic_cc <- coord_cartesian(ylim = c(-0.05, 0.2))

mf_pairs %>% 
      filter(! is.na(university.m)) %>% 
      ggplot(aes(university.m, EA3.f)) + 
        pic_ss + pic_cc +
        labs(x = "Male university attendance", y = "Female PSEA")

mf_pairs %>% 
      filter(! is.na(university.f)) %>% 
      ggplot(aes(university.f, EA3.m)) + 
        pic_ss + pic_cc +
        labs(x = "Female university attendance", y = "Male PSEA")
        
```


```{r pic-basic-corr-income, fig.subcap = c("Female EA3 by male income", "Male EA3 by female income"), fig.cap = "Social and genetic advantage among spouse pairs in UK Biobank", fig.width = 3, fig.height = 3}

pic_ss <- stat_summary(fun.data = mean_cl_normal, na.rm = TRUE)
pic_cc <- coord_cartesian(ylim = c(-0.15, 0.25))

mf_pairs %>% 
      filter(! is.na(income_cat.m)) %>% 
      ggplot(aes(income_cat.m, EA3.f)) + 
        pic_ss + pic_cc + 
        labs(x = "Male income category", y = "Female PSEA")

mf_pairs %>% 
      filter(! is.na(income_cat.f)) %>% 
      ggplot(aes(income_cat.f, EA3.m)) + 
        pic_ss + pic_cc +
        labs(x = "Female income category", y = "Male PSEA")
        
```


# Theory

There is a large population, whose members have a single genetic trait
$g_{i}$ and a single social trait $s_{i}$. 

Suppose that $G$ and $S$ are continuously distributed. Without loss
of generality, $EG = ES = 0$.^[Continuous distribution is not strictly required.
All that is needed is for a set of pairs of positive measure to have different values
of $G$ and $S$, along a set of attractiveness curves of positive measure.] People 
pair according to an attractiveness function

\[
A(g_{i},s_{i}) = f((1-k)g_{i},ks_{i})
\]

which is smooth and strictly increasing in both its arguments. If
$k = 0$, "indifference curves" of attractiveness are vertical lines
in $(G, S)$ space. If $k = 1$, they are horizontal lines. If $k \in (0,1)$
they are arbitrary downward sloping curves. 

Write $p(i)$ for $i$'s partner. Pairs always have the
same attractiveness.

\[
A(g_{i},s_{i}) = A(g_{p(i)},s_{p(i)}).
\]

Each pair has two children. We assume that both children $c(d)$ of
parents $d$, $m$ have

\begin{align}
g_{c(d)} & =\frac{g_{d}+g_{m}}{2};\label{eq:children-average}\\
s_{c(d)} & =\frac{s_{d}+s_{m}}{2}.\nonumber 
\end{align}

This is a strong assumption; we relax it later. For real world examples
approximated by it, $S$ could be wealth which is equally divided
between the children; $G$ could be a highly polygenic trait with
many small effects. Write $G_{p}, S_{p}$ to denote the population
variables in the parents' generation; $G_{c}, S_{c}$ for the children's
generation. 

\begin{prop}
\label{prop:prop1}
(i) $Cov(G_{c}, S_{c}) \ge Cov(G_{p},S_{p})$, with strict inequality
if and only if $0 < k < 1$. 

(ii) If $corr(G_{p}, S_{p}) \ge 0$, then $corr(G_{c},S_{c}) \ge corr(G_{p},S_{p})$,
with strict inequality if and only if $0 < k < 1$ \uline{or} 
$corr(G_{p}, S_{p}) > 0$.
\end{prop}


\begin{proof}

Within each pair $i,p(i)$ write $d$ for the person with $s_{d}>s_{p(d)}$
and $m$ for $p(d)$. (Think of these as "dukes" and "milkmaids",
or if you prefer "duchesses" and "tennis instructors".) If
$k < 1$, then $g_{d} < g_{m}$. (If $k = 0$, then define $d$ as the person
with $g_{d} < g_{p(d)}$.)

We integrate over the "dukes" to calculate the covariance in the
parents' generation:
\[
cov(G_{p},S_{p})=\int\frac{1}{2}(g_{d}s_{d}+g_{p(d)}s_{p(d)})\,\mathrm{d}d.
\]

For the children, the equivalent expression is
\[
cov(G_{c},S_{c})=\int g_{c(d)}s_{c(d)}\mathrm{d}d,
\]

observing that $EG_{c} = ES_{c}=0$ from (\ref{eq:children-average}).

Take an arbitrary pair $d,m$. Write

\begin{align*}
g_{d}s_{d} & =(g_{c}-\Delta g)(s_{c}+\Delta s);\\
g_{m}s_{m} & =(g_{c}+\text{\ensuremath{\Delta g)(s_{c}-\Delta s)}}
\end{align*}

where 

\begin{align*}
\text{\ensuremath{\Delta g=\frac{g_{m}-g_{d}}{2}\ge0}}, & \textrm{strictly so if and only if }k>0;\\
\Delta s=\frac{s_{d}-s_{m}}{2}\ge0 & \textrm{strictly so if and only if }k<1.
\end{align*}

Taking the average of the parents gives

\[
\frac{1}{2}(g_{d}s_{d}+g_{m}s_{m})= g _{c}s_{c}-\Delta g\Delta s.
\]

This is less than $g_{c}s_{c}$ if $0 < k < 1$, and equal to it if $k = 0$
or $k = 1$. Plugging this into the integral shows that 
\[
cov(G_{p},S_{p})\le cov(G_{c},S_{c})
\]
again with strict inequality if and only if $0 < k < 1$. This proves
the first part. A similar argument, showing $var(G_c) \le var(G_p)$ and 
$var(S_c) \le var(S_p)$, proves the second part (see the appendix).

\end{proof}


```{r pic-intuition, fig.subcap = c("Caste society (k = 1): parents", "Caste society (k = 1): children", "Egalitarian society (k = 0): parents", "Egalitarian society (k = 0): children", "Intermediate society (0 < k < 1): parents", "Intermediate society (0 < k < 1): children"), fig.cap = "Theory: shaded area is the population distribution. Dotted lines are attractiveness isoquants. Solid dots are example parents, transparent dots are example children. The right hand side shows the children's generation.", fig.ncol = 2, fig.width = 2.75, fig.height = 2.4, fig.align='center'}


setup_plot <- function () {
  par(mar = rep(2.1, 4))
  plot.new()
  plot.window(c(-1.2, 1.2), c(-1.2, 1.2))
  arrows(x0 = -1.1, x1 = 1.1, y0 = -1.1, length = .1)
  arrows(y0 = -1.1, y1 = 1.1, x0 = -1.1, length = .1)
  text(-1.1, 1.2, "S", cex = 0.8)
  text(1.2, -1.1, "G", cex = 0.8)
}

draw_dist <- function (w) {
  if (! is.matrix(w)) w <- matrix(c(w, 0, 0, 1), nrow = 2)
  mygrey <- grey(.85, .03)
  for (rad in seq(0.2, 1, 0.1)) {
    v <- car::ellipse(c(0, 0), shape = w, radius = rad, add = TRUE, center.pch = NULL,
               fill = TRUE, col = mygrey, lwd = 0, segments = 241)
  }
  
  v
}

draw_dist_with_a <- function (width, shift, curved_lines = FALSE) {
  setup_plot()  
  v <- draw_dist(width)
  nr_v <- nrow(v)
  
  rotation <- (-shift + seq_len(nr_v)) %% nr_v
  rotation[rotation == 0] <- nr_v
  
  v <- v[rotation, ]
  v_rev <- v[nr_v:1, ]
  
  ivls <- quantile(seq_len(nr_v/2), seq(0.2, 0.8, .2), type = 1)
  
  start_x <- v[ivls, 1]
  end_x   <- v_rev[ivls, 1]
  start_y <- v[ivls, 2]
  end_y <- v_rev[ivls, 2]
  
  if (curved_lines) {
    for (l in seq_along(start_x)) {
      sx <- start_x[l] + c(0, 0.15, 0.85, 1) * (end_x[l] - start_x[l])
      sy <- start_y[l] + c(0, 0.05, 0.65, 1) * (end_y[l] - start_y[l])
      spl <- stats::spline(sx, sy)
      if (l ==2) mid_spl <- spl
      lines(spl, lty = 2)
    }
    
    start_x <- mid_spl$x[2]
    start_y <- mid_spl$y[2]
    end_x <- mid_spl$x[8]
    end_y <- mid_spl$y[8]
    inset <- 0
  } else {
    segments(x0 = start_x, x1 = end_x, y0 = start_y, y1 = end_y, lty = 2)  
    
    ivls <- ivls[2:3]
  
    start_x <- mean(v[ivls, 1])
    end_x   <- mean(v_rev[ivls, 1])
    start_y <- mean(v[ivls, 2])
    end_y   <- mean(v_rev[ivls, 2])
    inset <- 0.15
  }
  
  
  along <- function (tt) list(
    x = start_x + tt * (end_x - start_x),
    y = start_y + tt * (end_y - start_y)
  )
  
  end_inset <- 0.15
  starts <- along(c(inset, 1 - inset))
  ends   <- along(c(0.5 - end_inset, 0.5 + end_inset))
  pts <- along(c(inset/2, .5, 1 - inset/2))
  arrows(x0 = starts$x, x1 = ends$x, y0 = starts$y, y1 = ends$y,
         length = 0.05)
  
  points(x = pts$x, y = pts$y, pch = c(19, 21, 19), cex = 0.8)
}

draw_dist_with_a(.8, 0)
setup_plot()
invisible(draw_dist(matrix(c(.4, 0, 0, 1), nrow = 2)))

draw_dist_with_a(.8, 60)
setup_plot()
invisible(draw_dist(matrix(c(.8, 0, 0, .4), nrow = 2)))

draw_dist_with_a(.8, -30, curved_lines = TRUE)
# setup_plot()
# invisible(draw_dist(matrix(c(1, .4, .4, .8), nrow = 2)))



```


Figure \@ref(fig:pic-intuition) shows the intuition behind this result. The top
row shows a *caste society* with $k = 1$. A typical pair is shown: children have
intermediate values of $G$ and $S$ between their two parents (hollow circle). In
this society pairs match only by social status; genetics plays no role. As a
result, while the variance of $G$ shrinks within each status group, genetics
remain uncorrelated with social status in the children's population
distribution, shown on the right. The next row shows an *egalitarian society*
with $k = 0$. Parents match only by genetics and ignore social status. Again, as
a result there is no correlation between genetics and social status in the
children's generation. The bottom row shows an intermediate society. Because
both genetics and social status contribute to attractiveness, matched spouses
typically trade them off against each other. As a result, the distribution is
squeezed along the gradient of $k$, and $G$ and $S$ are correlated in the children's 
generation.

Figure \ref{fig:Correlation-counterexample} in the Appendix shows that
the condition in the second part cannot be relaxed further.



## Robustness

We now relax the condition that children are exactly at the mean
of their parents' values for $G$ and $S$. Let

\begin{align*}
g_{c(i)} & =\bar{g}_{i}+\varepsilon_{i}^{G}\\
s_{c(i)} & =\bar{s}_{i}+\varepsilon_{i}^{S}
\end{align*}

where 

\[
\bar{g}_{i} = \frac{g_{i} + g_{p(i)}}{2}; \bar{s}_{i} = \frac{s_{i} + s_{p(i)}}{2}
\]

, $\varepsilon^{G}$ has mean 0 and variance $\sigma_{G}^{2}$ and
$\varepsilon^{S}$ has mean 0 and variance $\sigma_{S}^{2}$. 

\begin{prop}
\label{prop:robustness}
\begin{enumerate}
\item if $\sigma_{G}^{2}$ and $\sigma_{S}^{2}$ are small enough and $corr(G_{p},S_{p}) \ge 0$,
then $corr(G_{c},S_{c}) > corr(G_{p},S_{p})$ for $k \in (0,1)$.
\item if $\varepsilon^{G}$ and $\varepsilon^{S}$ are uncorrelated with
each other and with $\bar{G}$ and $\bar{S}$; and if $G_{p}$ and
$S_{p}$ are uncorrelated, then $corr(G_{c},S_{c}) \ge 0$, with strict
inequality if and only if $0<k<1$.
\end{enumerate}
\end{prop}

The conditions in Proposition \label{prop:robustness} are quite plausible. For
$G$, they require that either variance in siblings' scores on some summary statistic
is not too large, or that it is uncorrelated with the parents' scores. Both of 
these hold for most polygenic scores, which are additive sums of many small effects
of alleles derived randomly from one or other parent. For $S$, the conditions
would hold, for example, if $S$ measures wealth, which is inherited not too
unequally between siblings; or if wealth is inherited unequally but not in a
way that correlates with $S$ or $G$.

It is worth considering what kind of social arrangements would *violate* these
conditions. For example, suppose that parents' combined wealth is inherited by
the child with the lowest value of $g_{c(i)}$. This creates a negative correlation
between $s_{c(i)}$ and $g_{c(i)}$.



## Discussion

The "marriage market" here is a reduced form
mechanism, encompassing that makes a difference to partner choice. For example,
if earned income affects attractiveness in the marriage market, then society's
level of meritocracy in the labour market will correlate with the value of $k$:
a more meritocratic labour market will allow people with low social status but
high human capital (partly genetically determined) to earn more, and therefore
to enter the high group.

Also, the contents of $G$ -- what counts as "good genes" in the marriage
market -- are themselves likely to vary across societies. For instance,
standards of physical attractiveness vary historically. Similarly, it is
plausible that what counted as a "good match", in terms of personality, physical
and intellectual characteristics, differed between medieval European nobility
and contemporary society. 

The model predicts variation in the strength of genetic lock-in. In
particular, in "caste societies" where there is complete endogamy within 
social status groups, there is no scope for genetic lock-in, because
marriage partners do not trade off genetics for social status. The model also 
assumes that social status is inherited randomly from one parent, in 
the same way a genetic allele is inherited. This assumption can be weakened. 
For example, if social status is inherited deterministically from the father, then the
results remain unchanged (for each pair of parents, just assume that one randomly
chosen parent is the father). 

Behaviour geneticists often make the point that in meritocratic societies,
successful people may transmit relevant genes to their offspring. (TODO: cite
relevant papers.) Like genetic genetic lock-in, meritocracy may therefore lead
to a correlation between social status and genetics. However, genetic lock-in is
a distinct, though overlapping, mechanism. Under meritocracy, certain genetic variants *cause* higher social
status and are then transmitted along with it. This logic does not apply in
non-meritocratic societies where social status is ascribed rather than earned.
Conversely, genetic variations which cause social status will become associated
with it, even in the absence of assortative mating.

By contrast, genetic lock-in applies to genetic variants that are *associated*
with higher social status in the spouse matching process. They do not need to
exert any influence whatsoever on an individual's own social status. This process
requires assortative mating, but does not require meritocracy. The logic
of genetic lock-in therefore applies to a historically much wider range of
societies, including societies where social status is wholly ascribed or
inherited, such as aristocracies.

In modern societies, both assortative mating and meritocracy are likely to be 
at play. Genetic variants that cause (e.g.) higher income and wealth will be 
inherited along with components of social status such as inherited wealth, 
networks and cultural capital. At the same time, higher social status and
"good genes" will assort in the marriage market, even if that higher social
status is caused by purely environmental variation. Our empirical analysis shows
this latter process at work.



# Data

As mentioned above, simple correlations between one partner's social status
and the other partner's genetics do not prove that genetic lock-in is
taking place, because one's social status correlates with one's own genetics.
To demonstrate genetic lock-in, we therefore need a source of social
advantage which is exogenous to genetics. One possibility is *birth order*.
It is well known that earlier-born children receive more parental care and
have better life outcomes. (XXX is it? Go check.) On the other hand, early-
and late-born full siblings have the same *ex ante* expected genetic endowment.
^[This might not be the case, if parents' choice of whether to
have more children is endogenous to the genetic endowment of their earlier 
children. We will check for this below.] We can therefore use birth order
as an exogenous shock to social status.

We use data from UK Biobank, a study of about 500,000 individuals. 

* TODO: describe N for birth order, describe PSEA calculation.
* TODO: look at mechanisms by which birth order might affect university
* TODO: get IQ data, control for it
* TODO: subset to spouses with children
* TODO: overall index of social status?


# Results 

Ideally we would instrument social status with birth order. However, our measures of
social status are noisy and incomplete. For example, we know whether subjects went
to university, but not which university they went to, and we only have rough
categorical data on household income. Birth order likely affects both these and 
other measures of social advantage. So, an instrumental variables approach
would probably fall foul of the exclusion restriction. 

Instead, we conduct a mediation analysis, following the strategy of Heckman and
Pinto (2013). We first regress our measures of own social status (i.e. income
and education) on birth order. Then, we regress spouse's PSEA on birth order,
with and without controlling for social status. Under the assumption that birth
order is exogenous to own genetics, these regressions identify the effect of
birth order, plus other environmental variables that correlate with it, on own
social status and spouse's genetics. Also, if the estimated effect of birth
order on spouse's PSEA changes when social status is included, that is evidence
that social status mediates the effect of birth order.

* TODO: clarify the empirical model, you may need help....
* TODO: estimate individual income from job SIC codes? ASHE gives data


```{r calc-bo-first-stage}
mod_bo_1st <- list()
mod_bo_1st$uni    <- lm(university ~ n_older_sibs + factor(n_sibs), famhist, 
                    subset = n_sibs >= 2 & n_sibs <= 8)
mod_bo_1st$income <- lm(income_cat ~ n_older_sibs + factor(n_sibs), famhist, 
                    subset = n_sibs >= 2 & n_sibs <= 8)

pval_bo_1st <- map(mod_bo_1st, 
                 ~tidy(.x) %>% filter(term == "n_older_sibs") %>% pull(p.value)
               )
pval_bo_1st %<>% map(format, digits = 3)
```

Figures \@ref(fig:pic-bo-uni) and \@ref(fig:pic-bo-income) show the
relationship between birth order, university education and income, separately for
respondents with 1-3 siblings. We test this formally in a linear regression
controlling for family size, which may be correlated with parents'
characteristics including genetics. Birth order is negatively correlated with
both measures (among respondents with 1-7 siblings: university $p$ = 
`r pval_bo_1st$uni`, income $p$ = `r pval_bo_1st$income`).


```{r pic-bo-uni, fig.cap = "University attendance by birth order and family size", fig.height = 3}
famhist %>% 
      filter(n_sibs <= 4, n_sibs >= 2, ! is.na(n_older_sibs)) %>% 
      mutate(
        `Birth order` = n_older_sibs + 1,
        `Family size` = n_sibs
      ) %>% 
      group_by(`Birth order`, `Family size`) %>% 
      dplyr::summarize(University = mean(university, na.rm = TRUE)) %>% 
      ggplot(aes(`Birth order`, University)) + 
      geom_col() +
      facet_grid(cols = vars(`Family size`), labeller = label_both) + 
      ylim(0, 0.15) 

```


```{r pic-bo-income, fig.cap = "Income by birth order and family size", fig.height = 3}
famhist %>% 
      filter(
        n_sibs <= 4, 
        n_sibs >= 2, 
        ! is.na(n_older_sibs), 
        ! is.na(income_cat)
      ) %>% 
      mutate(
        `Birth order` = n_older_sibs + 1,
        `Family size` = n_sibs,
        `Income category` = factor(income_cat)
      ) %>% 
      ggplot(aes(`Birth order`, fill = `Income category`)) + 
      geom_bar(position = "fill") +
      facet_grid(cols = vars(`Family size`), labeller = label_both) + 
      scale_fill_brewer(palette = 2) + 
      labs(y = "Proportion")
```


Next we run regressions of spouse PSEA on birth order, university attendance
and income.

TODO: try to explain this better!


```{r tbl-bo-psea}

fml_bo_psea <- list()

fml_bo_psea[[1]] <- EA3.y ~ n_older_sibs.x + EA3.x | factor(n_sibs.x)
fml_bo_psea[[2]] <- EA3.y ~ university.x + n_older_sibs.x + EA3.x | factor(n_sibs.x)
fml_bo_psea[[3]] <- EA3.y ~ income_cat.x + n_older_sibs.x + EA3.x | factor(n_sibs.x)
fml_bo_psea[[4]] <- EA3.y ~ income_cat.x + university.x + n_older_sibs.x + EA3.x |
      factor(n_sibs.x)

mf_pairs_reg <- mf_pairs_twice %>% 
                  filter(
                    n_sibs.x >= 2, 
                    n_sibs.x <= 8, 
                    ! is.na(n_older_sibs.x),
                    ! is.na(university.x),
                    ! is.na(income_cat.x)
                  )

mod_bo_psea <- lapply(fml_bo_psea, fixest::feols, 
                 data = mf_pairs_reg,
                 notes = FALSE
               )

huxreg(mod_bo_psea, 
        coefs = c(
          "Birth order" = "n_older_sibs.x", 
          "University"  = "university.xTRUE", 
          "Income"      = "income_cat.x",
          "Own EA3"     = "EA3.x"
        ),
         note = "{stars}. Standard errors clustered by spouse pair.",
         tidy_args = list(conf.int = FALSE, cluster = list(mf_pairs_reg$couple_id))
       ) %>% 
       insert_row(after = 9, "Family size dummies", rep("Yes", 4)) %>% 
       set_number_format(2:9, -1, 4) %>% 
       set_caption("Regressions of spouse PSEA")
```


```{r calc-bootstrap-proportions, cache = TRUE}

n_boot <- 199

bo_prop_boot <- replicate(n_boot, {
  mf_pairs_resample <- mf_pairs_reg %>% sample_frac(1, replace = TRUE)
  tmp  <- fixest::feols(fml_bo_psea[[1]], data = mf_pairs_resample, notes = FALSE)
  tmp2 <- fixest::feols(fml_bo_psea[[2]], data = mf_pairs_resample, notes = FALSE)
  tmp3 <- fixest::feols(fml_bo_psea[[3]], data = mf_pairs_resample, notes = FALSE)
  tmp4 <- fixest::feols(fml_bo_psea[[4]], data = mf_pairs_resample, notes = FALSE)
  orig_coef <- coef(tmp)["n_older_sibs.x"]
  c(
    prop_uni  = coef(tmp2)["n_older_sibs.x"]/orig_coef,
    prop_inc  = coef(tmp3)["n_older_sibs.x"]/orig_coef,
    prop_both = coef(tmp4)["n_older_sibs.x"]/orig_coef
  )
})
```


```{r clean-bootstrap-proportions}
bo_prop_ci <- apply(bo_prop_boot, 1, quantile, probs = c(.975, .025))
# the percentage reduction:
bo_prop_ci[] <- sprintf("%.1f%%", (1 - bo_prop_ci) * 100)

# point estimates from main regressions:
orig_coef <- coef(mod_bo_psea[[1]])["n_older_sibs.x"]
bo_prop_est <- apply(bo_prop_boot, 1, mean)
bo_prop_est  <- sprintf("%.1f%%", (1 - bo_prop_est) * 100)

```


Table \@ref(tab:tbl-bo-psea) shows the results. Column 1 shows the effect of 
birth order controlling only for own PSEA and family size. It establishes
that earlier-born children have spouses with higher PSEA. The effect size
is small. This is to be expected, because (a) the effects of birth order on
university, income and (presumably) other variables are small, and (b) PSEA is
measured with a lot of error. We aim to test theory rather than estimating 
an effect size, so we focus more on statistical significance.

Column 2 includes university attendance. Column 3 includes income. Column 4
includes both. We estimate the percentage decrease in the effect of birth order
across the columns, along with 95% confidence intervals for this figure, by
running bootstraps (N = `r n_boot`).^[The sample percentage decrease calculated
from the figures in Table \@ref(tab:tbl-bo-psea) is not the correct estimate,
since $E(X/Y) \ne EX/EY$.] Including university attendance alone reduces 
the effect of birth order by  `r bo_prop_est[1]`
(CI `r bo_prop_ci[1, 1]` -- `r bo_prop_ci[2, 1]`). Including income alone 
reduces the effect of birth order by `r bo_prop_est[2]` (CI 
`r bo_prop_ci[1, 2]` -- `r bo_prop_ci[2, 2]`).  Including both decreases the 
effect by `r bo_prop_est[3]` (CI `r bo_prop_ci[1, 3]` --`r bo_prop_ci[2, 3]`).


## Robustness


```{r tbl-pgs-check}

check_cor <- function (score_name) {
  fml_check <- as.formula(paste(score_name, "~ factor(n_sibs) + n_older_sibs"))
  mod_check <- lm(fml_check, famhist, n_sibs >= 2 & n_sibs <= 8)
  mod_check %>% 
        tidy() %>% 
        filter(term == "n_older_sibs") %>% 
        mutate(score = score_name)
}

drake::loadd(score_names)
score_names <- paste0(score_names, "_resid")
check_result <- purrr::map_dfr(score_names, check_cor)

if (any(check_result$p.value < 0.1/33)) stop("Some check p values < 0.1/33; rewrite text!")

n_10 <- sum(check_result$p.value < 0.1)
max_coef <- max(abs(check_result$estimate))
```


Although all children of the same parents have the same polygenic scores in
expectation, it could still be possible that genetics correlates with birth
order within the sample. This could happen if parents select family size on the
basis of genetics. For example, if the first child had a phenotype reflecting a
high (or low) polygenic score, then that might affect the parents' decision to
have a second child. Alternatively, respondents might select into the sample on
the basis of a combination of birth order and genetics. We check this by
regressing 33 different polygenic scores on birth order, controlling for family
size.^[Polygenic scores were residualized on the first principal components of
the genetic data.] Table \@ref(tab:tbl-pgs-check) shows the results. No scores
were significant at $p < 0.10/33$. `r n_10` scores were significant at $p <
0.10$ (body mass index, conscientousness, and neuroticism). Coefficients were
never greater than 0.01 of a standard deviation. Table
\@ref(tab:tbl-bo-psea-robust) in the appendix reruns regressions controlling for
these scores. Results are almost unchanged.


# Conclusion

TODO: Write!

\FloatBarrier

# Appendix

## Second part of Proposition \@ref(prop:prop1)

\begin{proof}
Write

\begin{equation}
corr(G_{j},S_{j})=\frac{cov(G_{j},S_{j})}{\sqrt{var(G_{j})var(S_{j})}}\textrm{ for both generations }j\in\{p,c\}.\label{eq:corr-cov-var}
\end{equation}

where 

\begin{align*}
var(G_{p}) & =\frac{1}{2}\int g_{d}^{2}+g_{p(d)}^{2}\textrm{d}d;\\
var(G_{c}) & =\int g_{c(d)}^{2}\textrm{d}d.
\end{align*}

Much as before,
\begin{align*}
g_{d}^{2}+g_{m}^{2} & =(g_{c}-\Delta g)^{2}+(g_{c}+\Delta g)^{2}\\
 & =2g_{c}^{2}+2(\Delta g)^{2}\\
 & \ge2g_{c}^{2}.
\end{align*}

This shows that $var(G_{c})\le var(G_{p})$ and a similar argument
shows $var(S_{c})\le var(S_{p})$. Thus the covariance is higher (and
positive) in the children's generation, while the variances are lower.
Combining these ensures that 

\[
corr(G_{c},S_{c}) \ge corr(G_{p},S_{p}).
\]

Since for any $k$, either $var(G_{c}) < var(G_{p})$ or $var(S_{c}) < var(S_{p})$,
the only way to get strict equality for the above is if $k \in \{0,1\}$
and $cov(G_{c}, S_{c}) = cov(G_{p},S_{p}) = 0$.

\end{proof}

To show that the condition in the second part cannot be relaxed further,
consider the distribution in Figure \ref{fig:Correlation-counterexample}.
There is negative correlation in the parents' generation (the shaded
area). If $k = 1$ or is close enough to 1, then assortative mating along the dotted
lines will reduce the variance of $S$ along those lines, pushing the
distribution towards the darker central area, without
affecting the covariance. This will make the correlation more negative.
After repeated generations the horizontal variance within values of $G$ will almost disappear and the correlation will approach -1.

\begin{figure}
\begin{centering}
\includegraphics[width=0.5\columnwidth]{correlation-counterexample}
\par\end{centering}
\caption{Correlation counterexample\label{fig:Correlation-counterexample}}

\end{figure}

## Proposition \ref{prop:robustness}

\begin{proof}

Note that in proposition \ref{prop:prop1}, we took $g_{c(i)} = \bar{g}_{i}$ and $s_{c(i)} = \bar{s}_i$. Write 

\begin{align}
cov(G_{c},S_{c}) & =cov(\bar{G}+\varepsilon^{G},\bar{S}+\varepsilon^{S})\nonumber \\
 & = cov(\bar{G},\bar{S}) + cov(\varepsilon^{G},\bar{S}) + 
     cov(\bar{G},\varepsilon^{S}) + cov(\varepsilon^{G},\varepsilon^{S}).
 \label{eq:cov-with-error}
\end{align}

For any $X$ and $Y$, $cov(X, Y)$ is bounded by $\sqrt{var(X) var(Y)}$.
Plugging $\sigma_{G}^{2}$ and $\sigma_{S}^{2}$ into this formula
shows that under condition 1, $cov(G_{c},S_{c})$ will be arbitrarily
close to $cov(\bar{G},\bar{S})$. Similarly, writing

\[
var(G_{c}) = var(\bar{G}) + var(\varepsilon^{G}) + 2cov(\bar{G},\varepsilon^{G})
\]

shows that $var(G_{c})$ will approach $var(\bar{G})$ as $\sigma_{G}^{2}$
grows small, and similarly for $var(S_{c})$. Plugging these facts
into (\ref{eq:corr-cov-var}) shows that $corr(G_{c},S_{c})$ approaches
$corr(\bar{G},\bar{S})$ as $\sigma_{G}^{2}$ and $\sigma_{S}^{2}$
grow small. Proposition 1 then shows $corr(\bar{G},\bar{S}) < corr(G_{p},S_{p})$
for $k\in(0,1)$.

Under condition 2, $cov(G_{c}, S_{c}) = cov(\bar{G},\bar{S})$ since
the last three terms of the sum in (\ref{eq:cov-with-error}) are
zero. Then since 
\[
cov(\bar{G},\bar{S})\ge cov(G_{p},S_{p}) = 0
\]
with strict inequality iff $k\in(0,1)$, the covariance signs the
correlation. 

\end{proof}

\FloatBarrier

## Regressions controlling for polygenic scores

```{r tbl-bo-psea-robust}

fml_bo_psea_robust <- fml_bo_psea %>% 
                        purrr::map(Formula::as.Formula) %>% 
                        purrr::map(update, 
                           . ~ . + 
                           bmi_combined_resid.x + 
                           conscientiousness_resid.x + 
                           neuroticism_resid.x
                        )

mod_bo_psea_robust <-  purrr::map(fml_bo_psea_robust, fixest::feols, 
                         data = mf_pairs_reg,
                         notes = FALSE
                       )

huxreg(mod_bo_psea_robust, 
        coefs = c(
          "Birth order" = "n_older_sibs.x", 
          "University"  = "university.xTRUE", 
          "Income"      = "income_cat.x",
          "Own EA3"     = "EA3.x"
        ),
         note = "{stars}. Standard errors clustered by spouse pair.",
         tidy_args = list(conf.int = FALSE, cluster = mf_pairs_reg$couple_id)
       ) %>% 
       insert_row(after = 9, "Family size dummies", rep("Yes", 4)) %>% 
       insert_row(after = 10, "Polygenic score controls", rep("Yes", 4)) %>% 
       set_top_border(11, everywhere, 0) %>% 
       set_number_format(2:9, -1, 4) %>% 
       set_caption("Regressions of spouse PSEA with controls for polygenic scores")
```


